### ISSUE TITLE
CI Tests Failure with Release of Python 3.12.0-alpha.7

### ISSUE THREAD

**Issue by rmartin16** [beeware/briefcase#1165]
### Describe the bug

CI test failure [showed up](https://github.com/beeware/briefcase/actions/runs/4633929661/jobs/8199635365) today; [wasn't happening](https://github.com/beeware/briefcase/actions/runs/4623630954) on Python 3.12.0-alpha.6

```
==============================================================================
ERROR: tests/integrations/download/test_Download__file.py::test_content_connection_error
------------------------------------------------------------------------------
Downloading something.zip...

Traceback (most recent call last):
  File "D:\a\briefcase\briefcase\tests\integrations\download\test_Download__file.py", line 363, in test_content_connection_error
    mock_tools.download.file(
  File "D:\a\briefcase\briefcase\.tox\py\Lib\site-packages\briefcase\integrations\download.py", line 85, in file
    self._fetch_and_write_content(response, filename)
  File "D:\a\briefcase\briefcase\.tox\py\Lib\site-packages\briefcase\integrations\download.py", line 116, in _fetch_and_write_content
    temp_file.write(response.content)
  File "C:\hostedtoolcache\windows\Python\3.12.0-alpha.7\x64\Lib\tempfile.py", line 483, in func_wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
TypeError: a bytes-like object is required, not 'MagicMock'
```

### Steps to reproduce

CI

### Expected behavior

CI passes.

### Environment

- Operating System: Windows, macOS, Ubuntu
- Python version: 3.12.0-alpha.7
- Software versions:
  - Briefcase: `0.3.14.dev78+g3f8b8979`

**Comment by rmartin16** [comment]
well...I don't know what's causing this....maybe a bug in this alpha

fwiw, using a debug version of Python 3.12.0a7, I get a hard crash:
```
❯ pytest tests/integrations/download/ -k test_content_connection_error
Fatal Python error: Aborted

Current thread 0x00007f94a529fb80 (most recent call first):
  File "/home/russell/.pyenv/versions/3.12.0a7-debug/lib/python3.12/unittest/mock.py", line 647 in __getattr__
  File "/home/russell/github/beeware/briefcase/src/briefcase/integrations/download.py", line 116 in _fetch_and_write_content
  File "/home/russell/github/beeware/briefcase/src/briefcase/integrations/download.py", line 85 in file
  File "/home/russell/github/beeware/briefcase/tests/integrations/download/test_Download__file.py", line 361 in test_content_connection_error
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/python.py", line 195 in pytest_pyfunc_call
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/python.py", line 1789 in runtest
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 167 in pytest_runtest_call
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 260 in <lambda>
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 339 in from_call
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 259 in call_runtest_hook
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 220 in call_and_report
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 131 in runtestprotocol
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/runner.py", line 112 in pytest_runtest_protocol
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/main.py", line 349 in pytest_runtestloop
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/main.py", line 324 in _main
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/main.py", line 270 in wrap_session
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/main.py", line 317 in pytest_cmdline_main
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/config/__init__.py", line 167 in main
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/lib/python3.12/site-packages/_pytest/config/__init__.py", line 190 in console_main
  File "/home/russell/github/beeware/briefcase/venv-3.12.0a7-briefcase/bin/pytest", line 8 in <module>

Extension modules: markupsafe._speedups, psutil._psutil_linux, psutil._psutil_posix (total: 3)
Aborted (core dumped)
```

**Comment by freakboy3742** [comment]
Looks like a regression in [`unitest.mock.PropertyMock`](https://github.com/python/cpython/issues/103329).

**Comment by freakboy3742** [comment]
This has been fixed in CPython main in #103332; a regression test has been submitted in #103358. Unfortunately, we won't see the benefit of this until a new release is cut, which won't be until early May with the release of 3.12.0b1.

**Comment by rmartin16** [comment]
Python 3.12.0b1 included the fix for this issue.  Further work for Python 3.12 support is being tracked in #1290.
